### PR TITLE
Solving tests

### PR DESCRIPTION
**Pull request by ruleva1983**
## Description

- Disabled several tests in the auto-nested case.
- Re-enabled tests for SubTensorDicts due to change in previous merge
- Small modification in the apply_ function to allow for inplace argument

## Motivation and Context

Modification towards solving #116 .

## Types of changes

What types of changes does your code introduce? Remove all that do not apply:

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Tests modifications
- [x] Documentation (update in the documentation)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/pytorch/rl/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] My change requires a change to the documentation.
- [x] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: tensordict/tensordict.py [hunk 1]
```diff
@@ -615,18 +615,20 @@ def pop(
                 )
         return out
 
-    def apply_(self, fn: Callable) -> TensorDictBase:
+    def apply_(self, fn: Callable, inplace: bool = True) -> TensorDictBase:
```
Comment (tcbegley) [inline tensordict/tensordict.py:618]: It's a PyTorch convention that methods suffixed with `_` are inplace, so I don't think it makes sense to have the `inplace` keyword argument here.

File: test/test_tensordict.py [hunk 2]
```diff
@@ -733,40 +725,34 @@ def test_lock(self, td_name, device):
...
         td = td.select(inplace=True)
-        for key, item in td_clone.items(True):
+        keys_view = _TensorDictKeysView(td_clone, include_nested=True, leaves_only=False, error_on_loop=False)
+        for key in keys_view:
+            item = td_clone.get(key)
             with pytest.raises(RuntimeError, match="Cannot modify locked TensorDict"):
                 td.set(key, item)
         td.unlock()
-        for key, item in td_clone.items(True):
+        keys_view = _TensorDictKeysView(td_clone, include_nested=True, leaves_only=False, error_on_loop=False)
```
Comment (tcbegley) [inline test/test_tensordict.py:741]: I think you can reuse the `keys_view` you created above

File: test/test_tensordict.py [hunk 3]
```diff
@@ -1362,7 +1352,7 @@ def test_setitem(self, td_name, device, idx):
 
         td_clone = torch.cat([td_clone, td_clone], 0)
         with pytest.raises(RuntimeError, match="differs from the source batch size"):
-            td[idx] = td_clone
+           td[idx] = td_clone
```
Comment (tcbegley) [inline test/test_tensordict.py:1355]: This looks underindented to me

File: test/test_tensordict.py [hunk 4]
```diff
@@ -850,7 +836,7 @@ def test_masking(self, td_name, device):
         assert_allclose_td(td_masked, td_masked2)
         assert td_masked.batch_size[0] == mask.sum()
         assert td_masked.batch_dims == 1
-
+        #
```
Comment (tcbegley) [inline test/test_tensordict.py:839]:
```suggestion

```

File: test/test_tensordict.py [hunk 5]
```diff
@@ -805,13 +784,15 @@ def test_zero_(self, td_name, device):
     def test_apply(self, td_name, device, inplace):
         td = getattr(self, td_name)(device)
         td_c = td.to_tensordict()
-        td_1 = td.apply(lambda x: x + 1, inplace=inplace)
+        td_1 = td.apply_(lambda x: x + 1, inplace=inplace)
```
Comment (tcbegley) [inline test/test_tensordict.py:787]: `apply` and `apply_` are two different methods, this test is for `apply` (which with `inplace=True` will call `apply_`). If we change `apply` to `apply_` then we are losing test coverage

**Review by tcbegley — CHANGES_REQUESTED** [review]

File: tensordict/tensordict.py (hunk 1 above)
Comment (ruleva1983) [reply tensordict/tensordict.py:618]: Restored as before

File: test/test_tensordict.py (hunk 2 above)
Comment (ruleva1983) [reply test/test_tensordict.py:741]: Correct!!

File: test/test_tensordict.py (hunk 3 above)
Comment (ruleva1983) [reply test/test_tensordict.py:1355]: corrected

File: test/test_tensordict.py (hunk 5 above)
Comment (ruleva1983) [reply test/test_tensordict.py:787]: Reverted for now

**Review by tcbegley — APPROVED** [review]
LGTM, thanks!

**Pull request merged**